### PR TITLE
Fix broken links

### PR DESCRIPTION
**Pull request by stwunsch**
First batch fixing the most obvious broken links, not taking releases and release notes into account

### CODE REVIEW

File: install/build_from_source.md [hunk 1]
```diff
@@ -254,7 +254,7 @@ The user can set any CMake variable or option that controls the build process fr
 | qt | | Qt graphics backend, requires libqt >= 4.8 |
 | qtgsi | * | GSI's Qt integration, requires libqt >= 4.8 |
 | roofit | * | Build the libRooFit advanced fitting package |
-| root7 | OFF | ROOT 7 support ([read more](/root-7)) |
+| root7 | OFF | ROOT 7 support |
```
Comment (eguiraud) [inline install/build_from_source.md:257]: can you please make this a link to the ROOT7 section in this file?

Comment (stwunsch) [reply install/build_from_source.md:257]: There is none, right? @Axel-Naumann Did I get this wrong?

Comment (eguiraud) [reply install/build_from_source.md:257]: https://root-project.github.io/web/install/build_from_source/#enabling-experimental-features-aka-root7

Comment (stwunsch) [reply install/build_from_source.md:257]: This is just right above the table with the build option. I can add the link there, though it seems to me not necessary.

Comment (stwunsch) [reply install/build_from_source.md:257]: And it's not "read more", it says just that you required C++14. If you still vote for adding it, I'm happy to do so!

Comment (eguiraud) [reply install/build_from_source.md:257]: please :) it will contain more info (or links to more info) in the future

Comment (stwunsch) [reply install/build_from_source.md]: Done! :)

**Pull request merged**